src/server/service/kv.rs

```diff
@@ -767,7 +767,7 @@ impl<T: RaftStoreRouter<E::Local> + 'static, E: Engine, L: LockManager> Tikv for
 
         let region_id = req.get_context().get_region_id();
         let (cb, f) = paired_future_callback();
-        let mut split_keys = if req.is_raw_kv {
+        let mut split_keys = if req.is_raw_kv && self.storage.get_api_version() != ApiVersion::V2 {
```

Comment:
Suggest not to assume that API V2 has the same encoding with txn. Unless we have a unified encapsulation for both raw & txn data.

Reply:
done, use Api trat to handle raw keys.